### PR TITLE
Custom fields for db extractors

### LINKED ISSUE #2446: Add custom fields for DB extractors
There are a couple issues (sql server and redshift right now)  Where custom fields could be useful to solve them.  

ex: https://github.com/keboola/db-extractor-mssql/issues/86 and https://github.com/keboola/db-extractor-redshift/issues/14

### PR DESCRIPTION
**Pull request by pivnicek**
Fixes #2446 

Proposed changes:

-  add custom fields for db extractors (for import options etc)

### CODE REVIEW

**Comment by pivnicek** [comment]
@ujovlado I'm thinking of adding the refactor here to have custom field handling be the same for both credentials fields and these new ones.  If you agree, I'll go ahead with that.

**Comment by pivnicek** [comment]
@ErikZigo , doufam ze s timhle pak to bude docela jednoduchy vyresit https://github.com/keboola/db-extractor-redshift/issues/14

File: src/scripts/modules/ex-db-generic/react/components/QueryEditor.jsx [hunk 1]
```diff
@@ -8,13 +8,15 @@ import CodeEditor from '../../../../react/common/CodeEditor';
 import Select from '../../../../react/common/Select';
 import TableSelectorForm from '../../../../react/common/TableSelectorForm';
 import Tooltip from '../../../../react/common/Tooltip';
+import {Input} from './../../../../react/common/KbcBootstrap';
```
Comment (ujovlado) [inline src/scripts/modules/ex-db-generic/react/components/QueryEditor.jsx]: Please use components from react-bootstrap. There's a plan to get rid of KbcBootstrap "proxy"

**Comment by pivnicek** [comment]
@ujovlado I changed the newly introduced inputs to use straight react-bootstrap.  I think I should make a separate PR for converting this whole section to use react-bootstrap elements and consolidate the custom fields stuff into a single class.  What do you think?

**Comment by pivnicek** [comment]
@ujovlado there is two right now (one for mssql and one for redshift) and another one for mssql coming soon possibly.  It is generic enough right now that more are easy to add, I was just thinking that there is duplicate code with the credentials form this way.

**Comment by pivnicek** [comment]
regardless I'd like to keep it to a different PR as this option needs to go out asap

**Comment by ujovlado** [comment]
I don't get it, this PR was a mess.

- not at the top of the master (rebase was a PITA, because there was some merge commit with actual functionality)
- currently, there is only a checkbox in custom fields, standard text input doesn't have to be implemented at all (at this time)
- query creation from row wasn't there - so after save it didn't show as checked - even if it was save
- and one minor thing - Checkbox from react-bootstrap

**Comment by ujovlado** [comment]
@pivnicek check if it's working for you, I'll merge it. If I understand it correctly, this have to merged today.

File: src/scripts/modules/ex-db-generic/storeProvisioning.js [hunk 2]
```diff
@@ -41,9 +42,12 @@ export function queryFromRow(row) {
   } else {
     query = query.set('table', rowConfig.get('table', null));
   }
-  if (rowConfig.has('nolock')) {
-    query = query.set('nolock', rowConfig.get('nolock'));
-  }
+  // custom fields
+  getAllCustomFieldsNames().forEach((fieldName) => {
```
Comment (pivnicek) [inline src/scripts/modules/ex-db-generic/storeProvisioning.js]: wonderful

**Comment by ujovlado** [comment]
Not a big deal, but for change this small I didn't expect there'll be issues. :) It's good to keep things simple and not to worry about future usages (like that input) unless you need them.

**Comment by ujovlado** [comment]
So is it ok?

**Comment by ujovlado** [comment]
Oh, and last question. Is the `nolock` needed if query is in advanced mode? I assume not.

**Comment by pivnicek** [comment]
Actually, nolock is not-allowed for advanced mode:).  I should add that too.  But first I'm fixing https://github.com/keboola/db-extractor-mssql/issues/92

**Comment by pivnicek** [comment]
ok, made that fix, it's here if you are interested https://github.com/keboola/db-extractor-mssql/pull/93

I'm not sure what to do about the hiding/disabling this option for advanced queries.  What do you think?  Maybe disable it, I'll try it locally and see what it feels like.

**Comment by pivnicek** [comment]
nah, that's not going to work because it'll keep its value even if it's disabled.  Best to hide it and save it I guess.  Hmmm,  not sure.  What do you think @ujovlado ?

**Comment by pivnicek** [comment]
ok, I've got an idea to add a `mode` property to the custom field and save accordingly.

**Comment by pivnicek** [comment]
btw, @ujovlado, other than this minor thing, it works great.  worst case is if they set it for advanced mode as is it will result in a UserException.

**Comment by ujovlado** [comment]
> worst case is if they set it for advanced mode as is it will result in a UserException.

wouldn't be better to just ignore it in advanced mode?

**Comment by pivnicek** [comment]
Yes, probably:)

**Comment by pivnicek** [comment]
mmt
I'll make that last touch there on the backend.  jinak  this looks good, and thank you again

**Comment by pivnicek** [comment]
@ujovlado ok, it's ignored here: https://github.com/keboola/db-extractor-mssql/pull/94  If you could quickly have a look I'll deploy it and then we can merge this.

**Comment by pivnicek** [comment]
Might only be worth adding to the help text here (not applicable for advanced queries) I dunno.

**Comment by ujovlado** [comment]
OK, last update. I'm going to merge it.

**Pull request merged**